<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ProfileManager</class>
 <widget class="QDialog" name="ProfileManager">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>500</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Profile Management</string>
  </property>
  <property name="styleSheet">
   <string notr="true">/* ===== Modern App Styling with Animations ===== */
QWidget {
    background: qlineargradient(x1:0, y1:0, x2:1, y2:1,
        stop:0 #0a0a1a, stop:0.3 #1a1a2e, stop:0.7 #16213e, stop:1 #0f0f23);
    font-family: 'Segoe UI', 'SF Pro Display', 'Arial', sans-serif;
    color: #ffffff;
}

/* ===== Animated Background Elements ===== */
QWidget::before {
    content: "";
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background: 
        radial-gradient(circle at 20% 80%, rgba(0, 180, 216, 0.08) 0%, transparent 50%),
        radial-gradient(circle at 80% 20%, rgba(92, 107, 192, 0.08) 0%, transparent 50%),
        radial-gradient(circle at 40% 40%, rgba(255, 179, 0, 0.04) 0%, transparent 50%);
    animation: backgroundShift 25s ease-in-out infinite;
}

/* ===== Modern Input Fields ===== */
QLineEdit {
    background: rgba(255, 255, 255, 0.08);
    backdrop-filter: blur(10px);
    color: #ffffff;
    padding: 16px 20px;
    border-radius: 16px;
    border: 1px solid rgba(255, 255, 255, 0.1);
    font-size: 15px;
    font-weight: 400;
    box-shadow: 
        0 4px 16px rgba(0, 0, 0, 0.1),
        inset 0 1px 0 rgba(255, 255, 255, 0.1);
    transition: all 0.3s cubic-bezier(0.4, 0, 0.2, 1);
}

QLineEdit:focus {
    background: rgba(255, 255, 255, 0.12);
    border: 1px solid rgba(0, 212, 255, 0.5);
    box-shadow: 
        0 0 0 4px rgba(0, 212, 255, 0.1),
        0 8px 24px rgba(0, 212, 255, 0.2),
        inset 0 1px 0 rgba(255, 255, 255, 0.2);
    transform: translateY(-2px);
}

QLineEdit::placeholder {
    color: rgba(255, 255, 255, 0.5);
    font-style: normal;
    font-weight: 400;
}

QLineEdit[readOnly="true"] {
    background: rgba(255, 255, 255, 0.04);
    border: 1px solid rgba(255, 255, 255, 0.05);
    color: rgba(255, 255, 255, 0.7);
}

/* ===== Modern Text Edit ===== */
QTextEdit {
    background: rgba(255, 255, 255, 0.08);
    backdrop-filter: blur(10px);
    color: #ffffff;
    padding: 16px 20px;
    border-radius: 16px;
    border: 1px solid rgba(255, 255, 255, 0.1);
    font-size: 15px;
    font-weight: 400;
    box-shadow: 
        0 4px 16px rgba(0, 0, 0, 0.1),
        inset 0 1px 0 rgba(255, 255, 255, 0.1);
    transition: all 0.3s cubic-bezier(0.4, 0, 0.2, 1);
}

QTextEdit:focus {
    background: rgba(255, 255, 255, 0.12);
    border: 1px solid rgba(0, 212, 255, 0.5);
    box-shadow: 
        0 0 0 4px rgba(0, 212, 255, 0.1),
        0 8px 24px rgba(0, 212, 255, 0.2),
        inset 0 1px 0 rgba(255, 255, 255, 0.2);
    transform: translateY(-2px);
}

/* ===== Modern Date Edit ===== */
QDateEdit {
    background: rgba(255, 255, 255, 0.08);
    backdrop-filter: blur(10px);
    color: #ffffff;
    padding: 16px 20px;
    border-radius: 16px;
    border: 1px solid rgba(255, 255, 255, 0.1);
    font-size: 15px;
    font-weight: 400;
    box-shadow: 
        0 4px 16px rgba(0, 0, 0, 0.1),
        inset 0 1px 0 rgba(255, 255, 255, 0.1);
    transition: all 0.3s cubic-bezier(0.4, 0, 0.2, 1);
}

QDateEdit:focus {
    background: rgba(255, 255, 255, 0.12);
    border: 1px solid rgba(0, 212, 255, 0.5);
    box-shadow: 
        0 0 0 4px rgba(0, 212, 255, 0.1),
        0 8px 24px rgba(0, 212, 255, 0.2),
        inset 0 1px 0 rgba(255, 255, 255, 0.2);
    transform: translateY(-2px);
}

QDateEdit::drop-down {
    border: none;
    width: 24px;
}

QDateEdit::down-arrow {
    image: none;
    border-left: 6px solid transparent;
    border-right: 6px solid transparent;
    border-top: 6px solid #ffffff;
    margin-right: 8px;
}

/* ===== Modern Buttons with Animations ===== */
QPushButton {
    background: qlineargradient(x1:0, y1:0, x2:0, y2:1,
        stop:0 rgba(0, 180, 216, 0.9), stop:1 rgba(0, 150, 199, 0.9));
    color: white;
    border: none;
    border-radius: 16px;
    padding: 16px 24px;
    font-weight: 600;
    font-size: 14px;
    text-transform: uppercase;
    letter-spacing: 0.5px;
    box-shadow: 
        0 8px 24px rgba(0, 180, 216, 0.3),
        0 0 0 1px rgba(255, 255, 255, 0.1);
    transition: all 0.3s cubic-bezier(0.4, 0, 0.2, 1);
    position: relative;
    overflow: hidden;
}

QPushButton::before {
    content: "";
    position: absolute;
    top: 0;
    left: -100%;
    width: 100%;
    height: 100%;
    background: linear-gradient(90deg, transparent, rgba(255, 255, 255, 0.2), transparent);
    transition: left 0.5s;
}

QPushButton:hover {
    background: qlineargradient(x1:0, y1:0, x2:0, y2:1,
        stop:0 rgba(0, 212, 255, 0.9), stop:1 rgba(0, 180, 216, 0.9));
    transform: translateY(-4px) scale(1.02);
    box-shadow: 
        0 12px 32px rgba(0, 180, 216, 0.4),
        0 0 0 1px rgba(255, 255, 255, 0.2);
}

QPushButton:hover::before {
    left: 100%;
}

QPushButton:pressed {
    background: qlineargradient(x1:0, y1:0, x2:0, y2:1,
        stop:0 rgba(0, 150, 199, 0.9), stop:1 rgba(0, 120, 180, 0.9));
    transform: translateY(-2px) scale(0.98);
    box-shadow: 
        0 4px 16px rgba(0, 180, 216, 0.3),
        0 0 0 1px rgba(255, 255, 255, 0.1);
}

/* ===== Special Button Styling ===== */
QPushButton#pushButton_changePicture {
    background: qlineargradient(x1:0, y1:0, x2:0, y2:1,
        stop:0 rgba(78, 205, 196, 0.9), stop:1 rgba(68, 160, 141, 0.9));
    box-shadow: 
        0 8px 24px rgba(78, 205, 196, 0.3),
        0 0 0 1px rgba(255, 255, 255, 0.1);
}

QPushButton#pushButton_changePicture:hover {
    background: qlineargradient(x1:0, y1:0, x2:0, y2:1,
        stop:0 rgba(110, 231, 223, 0.9), stop:1 rgba(78, 205, 196, 0.9));
    box-shadow: 
        0 12px 32px rgba(78, 205, 196, 0.4),
        0 0 0 1px rgba(255, 255, 255, 0.2);
}

QPushButton#pushButton_deleteAccount {
    background: qlineargradient(x1:0, y1:0, x2:0, y2:1,
        stop:0 rgba(255, 107, 107, 0.9), stop:1 rgba(238, 90, 82, 0.9));
    box-shadow: 
        0 8px 24px rgba(255, 107, 107, 0.3),
        0 0 0 1px rgba(255, 255, 255, 0.1);
}

QPushButton#pushButton_deleteAccount:hover {
    background: qlineargradient(x1:0, y1:0, x2:0, y2:1,
        stop:0 rgba(255, 142, 142, 0.9), stop:1 rgba(255, 107, 107, 0.9));
    box-shadow: 
        0 12px 32px rgba(255, 107, 107, 0.4),
        0 0 0 1px rgba(255, 255, 255, 0.2);
}

QPushButton#pushButton_save {
    background: qlineargradient(x1:0, y1:0, x2:0, y2:1,
        stop:0 rgba(92, 107, 192, 0.9), stop:1 rgba(63, 81, 181, 0.9));
    box-shadow: 
        0 8px 24px rgba(92, 107, 192, 0.3),
        0 0 0 1px rgba(255, 255, 255, 0.1);
}

QPushButton#pushButton_save:hover {
    background: qlineargradient(x1:0, y1:0, x2:0, y2:1,
        stop:0 rgba(121, 134, 203, 0.9), stop:1 rgba(92, 107, 192, 0.9));
    box-shadow: 
        0 12px 32px rgba(92, 107, 192, 0.4),
        0 0 0 1px rgba(255, 255, 255, 0.2);
}

/* ===== Modern Labels ===== */
QLabel {
    color: #ffffff;
    font-weight: 500;
}

QLabel#label_profilePicture {
    background: rgba(255, 255, 255, 0.05);
    backdrop-filter: blur(20px);
    border: 2px solid rgba(255, 255, 255, 0.1);
    border-radius: 20px;
    color: rgba(255, 255, 255, 0.7);
    font-size: 15px;
    font-weight: 500;
    box-shadow: 
        0 12px 40px rgba(0, 0, 0, 0.3),
        0 0 0 1px rgba(255, 255, 255, 0.05),
        inset 0 1px 0 rgba(255, 255, 255, 0.1);
    transition: all 0.3s cubic-bezier(0.4, 0, 0.2, 1);
}

QLabel#label_profilePicture:hover {
    background: rgba(255, 255, 255, 0.08);
    border: 2px solid rgba(0, 180, 216, 0.3);
    transform: translateY(-2px);
    box-shadow: 
        0 16px 48px rgba(0, 0, 0, 0.4),
        0 0 0 1px rgba(0, 180, 216, 0.1),
        inset 0 1px 0 rgba(255, 255, 255, 0.15);
}

/* ===== Modern Scrollbars ===== */
QScrollBar:vertical {
    background: rgba(255, 255, 255, 0.05);
    width: 8px;
    border-radius: 4px;
    margin: 0;
}

QScrollBar::handle:vertical {
    background: qlineargradient(x1:0, y1:0, x2:1, y2:0,
        stop:0 rgba(0, 180, 216, 0.8), stop:1 rgba(0, 150, 199, 0.8));
    border-radius: 4px;
    min-height: 30px;
    margin: 2px;
}

QScrollBar::handle:vertical:hover {
    background: qlineargradient(x1:0, y1:0, x2:1, y2:0,
        stop:0 rgba(0, 212, 255, 0.9), stop:1 rgba(0, 180, 216, 0.9));
}

QScrollBar::add-line:vertical, QScrollBar::sub-line:vertical {
    height: 0px;
}

/* ===== Enhanced Tooltips ===== */
QToolTip {
    background: rgba(45, 45, 68, 0.95);
    backdrop-filter: blur(10px);
    color: #ffffff;
    border: 1px solid rgba(255, 255, 255, 0.1);
    border-radius: 12px;
    padding: 12px 16px;
    font-size: 13px;
    font-weight: 500;
    box-shadow: 
        0 8px 24px rgba(0, 0, 0, 0.3),
        0 0 0 1px rgba(255, 255, 255, 0.05);
}

/* ===== Animation Keyframes ===== */
@keyframes backgroundShift {
    0%, 100% {
        background-position: 0% 50%;
    }
    50% {
        background-position: 100% 50%;
    }
}

@keyframes float {
    0%, 100% {
        transform: translateY(0px);
    }
    50% {
        transform: translateY(-10px);
    }
}

@keyframes pulse {
    0%, 100% {
        opacity: 1;
    }
    50% {
        opacity: 0.8;
    }
}

/* ===== Responsive Design Elements ===== */
@media (max-width: 768px) {
    QLineEdit, QTextEdit, QDateEdit {
        padding: 14px 16px;
        font-size: 14px;
    }
    
    QPushButton {
        padding: 14px 20px;
        font-size: 13px;
    }
}
</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QLabel" name="label_profilePicture">
       <property name="minimumSize">
        <size>
         <width>150</width>
         <height>150</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>150</width>
         <height>150</height>
        </size>
       </property>
       <property name="frameShape">
        <enum>QFrame::Shape::Box</enum>
       </property>
       <property name="text">
        <string>Profile Picture</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignmentFlag::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QPushButton" name="pushButton_changePicture">
         <property name="text">
          <string>Change Picture</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_deleteAccount">
         <property name="text">
          <string>Delete Account</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QLineEdit" name="lineEdit_username">
     <property name="readOnly">
      <bool>true</bool>
     </property>
     <property name="placeholderText">
      <string>Username</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLineEdit" name="lineEdit_fullName">
     <property name="placeholderText">
      <string>Full Name</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLineEdit" name="lineEdit_email">
     <property name="placeholderText">
      <string>Email</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLineEdit" name="lineEdit_phone">
     <property name="placeholderText">
      <string>Phone</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QTextEdit" name="textEdit_address">
     <property name="placeholderText">
      <string>Address</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QDateEdit" name="dateEdit_dob"/>
   </item>
   <item>
    <widget class="QPushButton" name="pushButton_save">
     <property name="text">
      <string>Save Changes</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
